### PR TITLE
Minimal changes to work in Daml 2.0

### PR DESCRIPTION
**Pull request by arobert-da**
I needed an example of 2.0 ledger API access for another project so I modified this example, starting from #13.

* Daml script updated to use preferred syntax
* Because Daml parties include a globally-unique ID portion in 2.0, users and parties are created in the script (not `daml.yaml` and the UserManagementService is used to fetch the parties based on usernames)
* Package ID passing added for Reactive version comes from #13
* Application sleep lengthened to account for slower operation with Canton layer under 2.0

EDIT: Marking this PR as Draft because while the gRPC version runs similarly to on master, the output for the Reactive version is jumbled up at first and then deadlocks. I’ve included the ouput and the suspicious threads from the thread dump in the attached file.

[java-bindings-problems.txt](https://github.com/digital-asset/ex-java-bindings/files/8443498/java-bindings-problems.txt)

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -22,7 +22,7 @@
         </dependency>
         <dependency>
             <groupId>com.daml</groupId>
-            <artifactId>daml-lf-1.8-archive-java-proto</artifactId>
+            <artifactId>daml-lf-1.14-archive-java-proto</artifactId>
```
Comment (arobert-da) [inline pom.xml:25]: I'm not sure if this should be a 2.x version, it was taken from the original PR and seems to work.

Comment (realvictorprm) [reply pom.xml:25]: I think this is right but not exactly sure. Still investigating.

**Comment by realvictorprm** [comment]
so using a locally adjusted snapshot version where I reverted a commit I was able to get rid of the deadlocks, however it still didn't work correctly. After some investigation I found out that not all transactions are streamed correctly and I suspect now that there is a bug in the TransactionClientImpl.

**Comment by realvictorprm** [comment]
so using a locally adjusted snapshot version where I reverted a commit I was able to get rid of the deadlocks, however it still didn't work correctly. After some investigation I found out that not all transactions are streamed correctly and I suspect now that there is a bug in the TransactionClientImpl.

**Comment by arobert-da** [comment]
Incorporated into https://github.com/digital-asset/ex-java-bindings/pull/15 (merged)

**Pull request closed without merging**